src/box/schema.cc

```diff
@@ -124,7 +124,11 @@ box_schema_needs_upgrade(void)
 				    version_id_major(dd_version_id),
 				    version_id_minor(dd_version_id),
 				    version_id_patch(dd_version_id)),
-			 tarantool_version());
+			"",
```

Comment:
This function is now used only in Lua and the error is needed solely to log the warning. Let's drop the `diag_set()` and log the warning directly, as we did before commit 97c2c9a4fb58b:

https://github.com/tarantool/tarantool/blob/8ae450077341e53eda0f59acfbf044d5525f2196/src/box/lua/load_cfg.lua#L1150-L1157

Then we could make the error message a bit prettier:

```C
_(ER_SCHEMA_NEEDS_UPGRADE, 271, \
  "Your schema version is %s while Tarantool feature %s requires schema version %s or higher. " \
  "Please, consider using box.schema.upgrade().", \
  "schema_version", STRING, "feature_name", STRING, "required_schema_version", STRING) \
```

Reply:
Sure